iroh/src/node.rs

```diff
@@ -954,6 +955,121 @@ impl<D: BaoStore> RpcHandler<D> {
         rx.into_stream().map(BlobAddPathResponse)
     }
 
+    fn doc_import_file(
+        self,
+        msg: DocImportFileRequest,
+    ) -> impl Stream<Item = DocImportFileResponse> {
+        // provide a little buffer so that we don't slow down the sender
+        let (tx, rx) = flume::bounded(32);
+        let tx2 = tx.clone();
+        self.rt().local_pool().spawn_pinned(|| async move {
+            if let Err(e) = self.doc_import_file0(msg, tx).await {
+                tx2.send_async(DocImportProgress::Abort(e.into()))
+                    .await
+                    .ok();
+            }
+        });
+        rx.into_stream().map(DocImportFileResponse)
+    }
+
+    async fn doc_import_file0(
+        self,
+        msg: DocImportFileRequest,
+        progress: flume::Sender<DocImportProgress>,
+    ) -> anyhow::Result<()> {
+        use iroh_bytes::store::ImportMode;
+        use std::collections::BTreeMap;
+
+        let progress = FlumeProgressSender::new(progress);
+        let names = Arc::new(Mutex::new(BTreeMap::new()));
+        // convert import progress to provide progress
+        let import_progress = progress.clone().with_filter_map(move |x| match x {
+            ImportProgress::Found { id, name } => {
+                names.lock().unwrap().insert(id, name);
+                None
+            }
+            ImportProgress::Size { id, size } => {
+                let name = names.lock().unwrap().remove(&id)?;
+                Some(DocImportProgress::Found { id, name, size })
+            }
+            ImportProgress::OutboardProgress { id, offset } => {
+                Some(DocImportProgress::Progress { id, offset })
+            }
+            ImportProgress::OutboardDone { hash, id } => {
+                Some(DocImportProgress::IngestDone { hash, id })
+            }
+            _ => None,
+        });
+        let DocImportFileRequest {
+            doc_id,
+            author_id,
+            key,
+            path: root,
+            // todo: figure out tags
+            // in_place,
+            // tag,
+        } = msg;
+        // Check that the path is absolute and exists.
+        anyhow::ensure!(root.is_absolute(), "path must be absolute");
+        anyhow::ensure!(
+            root.exists(),
+            "trying to add missing path: {}",
+            root.display()
+        );
+
+        // todo: do we assume copy?
```

Comment:
I think we should expose the import mode to the rpc & client. Otherwise docs are second-class citizens with reagard to dealing with "big data" compared to blobs imported via Blobs rpc.

Reply:
great, will add this in